ugvc/mrd/featuremap_consensus_utils.py

```diff
@@ -0,0 +1,211 @@
+
+    Inputs:
+        record_dict (dict): A dictionary containing the record information
+        rec_id (str): The record id
+        out_fh (pysam.VariantFile): A pysam file handle of the output vcf file
+        header (pysam.VariantHeader): A pysam VarianyHeader pbject of the output vcf file
+        min_qual (int): The minimum quality threshold
+        qual_agg_func (function): The function to aggregate the quality scores (default: np.max)
+        sample_name (str): The name of the sample (default: "SAMPLE")
+    Output:
+        rec (pysam.VariantRecord): A pysam VariantRecord object
+
+    """
+    format_fields = [GT, AD, DP, VAF]
+
+    rec = header.new_record()
+    rec.chrom = rec_id[0]
+    rec.pos = rec_id[1]
+    rec.ref = rec_id[2]
+    rec.alts = rec_id[3]
+    rec.id = "."
+    rec.qual = qual_agg_func(record_dict[FeatureMapFields.X_QUAL.value])
+    # INFO fields to aggregate
+    # exceptions: X_QUAL
+    rec.info[FeatureMapFields.X_QUAL.value] = list(record_dict[FeatureMapFields.X_QUAL.value])
+    for field in fields_to_collect["numeric_array_fields"]:
+        if field in record_dict:
+            rec.info[field] = list(record_dict[field])
+    for field in fields_to_collect["string_list_fields"]:
+        if field in record_dict:
+            rec.info[field] = "|".join(record_dict[field])
+
+    # FORMAT fields to aggregate
+    record_dict[VAF] = record_dict[FeatureMapFields.FILTERED_COUNT.value] / record_dict[DP]
+    record_dict[GT] = (0, 1)
+    record_dict[AD] = [
+        record_dict[DP] - record_dict[FeatureMapFields.FILTERED_COUNT.value],
+        record_dict[FeatureMapFields.FILTERED_COUNT.value],
+    ]
+    for format_field in format_fields:
+        rec.samples[sample_name][format_field] = record_dict[format_field]
+    # FILTER field
+    if rec.qual < min_qual:
+        curr_filter = "LowQual"
+    elif rec.qual >= min_qual and record_dict[FeatureMapFields.FILTERED_COUNT.value] == 1:
+        curr_filter = "SingleRead"
+    else:
+        curr_filter = "PASS"
+    rec.filter.add(curr_filter)
+    # write to file
+    out_fh.write(rec)
+    return rec
+
+
+def pileup_featuremap(
+    featuremap: str, output_vcf: str, genomic_interval: str = None, min_qual: int = 0, sample_name: str = "SAMPLE"
+):
+    """
+    Pileup featuremap vcf to a reagular vcf, save the mean quality and read count
+
+    Inputs:
+        featuremap (str): The input featuremap vcf file
+        output_vcf (str): The output pileup vcf file
+        genomic_interval (str): The genomic interval to pileup, format: chr:start-end
+        min_qual (int): The minimum quality threshold
+        sample_name (str): The name of the sample (default: "SAMPLE")
+    Output:
+        output_vcf (str): The output vcf file
+    """
+    cons_dict = defaultdict(dict)
+    # process genomic interval input
+    if genomic_interval is None:
+        chrom = None
+        start = None
+        end = None
+    if genomic_interval is not None:
```

Comment:
"if genomic_interval is not None:" -> "else"

Please add this check:
assert re.match(r'^\w+:\d+-\d+$', genomic_interval), "Input genomic_interval format should be 'chrom:start-end'"